docs/learn/documentation/versioned/azure/eventhubs.md

```diff
@@ -0,0 +1,105 @@
+   See the License for the specific language governing permissions and
+   limitations under the License.
+-->
+
+You can configure your Samza jobs to process data from [Azure Eventhubs](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-features), Microsoft's data streaming service. An `event hub` is similar to a Kafka topic and can have multiple partitions with producers and consumers. Each message produced or consumed from an event hub is an instance of [EventData](https://docs.microsoft.com/en-us/java/api/com.microsoft.azure.eventhubs._event_data). 
+
+### Consuming from EventHubs:
+
+Samza's EventHubSystemConsumer wraps the EventData into an EventHubIncomingMessageEnvelope. The key of the message is set to the partition key of the EventData. If that is not present, then it will be set to the user defined `key` property map of the EventData. The message is obtained from the EventData body. 
+
+To configure Samza to configure from EventHub streams: 
+
+```
+# define an event hub system factory with your identifier. eg: eh-system
+systems.eh-system.samza.factory=org.apache.samza.system.eventhub.EventHubSystemFactory
+
+# define your streams
+systems.eh-system.stream.list=input0, output0
+
+# define required properties for your streams
+systems.eh-system.streams.input0.eventhubs.namespace=YOUR-STREAM-NAMESPACE
+systems.eh-system.streams.input0.eventhubs.entitypath=YOUR-ENTITY-NAME
+systems.eh-system.streams.input0.eventhubs.sas.keyname=YOUR-SAS-KEY-NAME
+systems.eh-system.streams.input0.eventhubs.sas.token=YOUR-SAS-KEY-TOKEN
+
+systems.eh-system.streams.output0.eventhubs.namespace=YOUR-STREAM-NAMESPACE
+systems.eh-system.streams.output0.eventhubs.entitypath=YOUR-ENTITY-NAME
+systems.eh-system.streams.output0.eventhubs.sas.keyname=YOUR-SAS-KEY-NAME
+systems.eh-system.streams.output0.eventhubs.sas.token=YOUR-SAS-KEY-TOKEN
+```
+
+The tuple required to access the Eventhubs entity per stream must be provided, namely the fields `YOUR-STREAM-NAMESPACE`, `YOUR-ENTITY-NAME`, `YOUR-SAS-KEY-NAME`, `YOUR-SAS-KEY-TOKEN`.
+
+### Producing to EventHubs:
+
+Similarly, you can also configure your Samza job to write to EventHubs.  
+```
+OutgoingMessageEnvelope envelope = new OutgoingMessageEnvelope(new SystemStream("eh-system", "output0"), key, message);
+collector.send(envelope);
+```
+The first parameter in the `OutgoingMessageEnvelope` is the name of the system and the output stream whose properties are specified in config. The `key` field in the envelope is used as the key in the `EventData` while the`message` field is used as its body.
+
+The envelopes sent will be converted to an `EventData` that is accepted by EventHub. The envelope `message` will be the body of the `EventData` and the envelope "key" and "produce time stamp" will be included in the properties map as `key` and `produce-timestamp` respectively. Currently there are no ways to pass other metadata in the properties field of `EventData`.
```

Comment:
Each [@link OutgoingMessageEnvelope] is converted into an [@link EventData instance] whose body is set to the message in the envelope. Additionally, the key and the produce timestamp are set as properties in the EventData before sending it to EventHubs.